lib/components/ApproveReject.js

```diff
@@ -67,76 +77,147 @@ class ApproveReject extends PureComponent {
+    let { proposalInfo, proposalValue, proposalRecipient } = this.props;
+
+    // add properties that may be missing
+    proposalInfo = this.makeProposalSafe(proposalInfo);
+
+    // hide based on selected proposal
+    let style = { visibility: 'hidden' };
+    if (proposalInfo && 'id' in proposalInfo) style = {};
+
+    return (
+      <BoxGrid rowClass="p-sm-2" style={style}>
+        <Label className="d-flex align-items-start" text="Cosigner Token">
+          <Input
+            className="col"
+            name="cosignerToken"
+            placeholder="Cosigner Token"
+            onChange={e => this.update('cosignerToken', e.target.value)}
+          />
+        </Label>
+        <Label className="d-flex align-items-start" text="Hardware Type">
+          <Dropdown
+            options={HARDWARE_TYPES}
+            onChange={({ value }) => this.update('selectedHardware', value)}
+          />
+        </Label>
+        <Label className="d-flex align-items-start" text="Account Number">
+          <Input
+            type="number"
+            name="accountNumber"
+            className="col"
+            placeholder="Account"
+            onChange={e => this.update('accountNumber', e.target.value)}
+          />
+        </Label>
+        <BoxGrid childrenClass="p-2 d-flex flex-column" colcount={2}>
+          <div>
+            <Label className="d-flex align-items-start" text="Creator ID">
+              <Text>{proposalInfo.author.name}</Text>
+            </Label>
+            <Label className="d-flex align-items-start" text="Amount">
+              <Text>{proposalValue}</Text>
+            </Label>
+            <Label className="d-flex align-items-start" text="Recipient">
+              <Text type="condensed">{proposalRecipient}</Text>
+            </Label>
+          </div>
+          <div>
+            <Label className="d-flex align-items-start" text="Approvals">
+              <Text>{proposalInfo.approvals.length}</Text>
+            </Label>
+            <Label className="d-flex align-items-start" text="Rejections">
+              <Text>{proposalInfo.rejections.length}</Text>
+            </Label>
+            <Label className="d-flex align-items-start" text="Status">
+              <Text>{proposalInfo.statusMessage}</Text>
+            </Label>
+          </div>
+        </BoxGrid>
+        <div className="d-flex flex-row justify-content-center">
+          <div className="flex-grow-1" />
+          <Button className="p-2 m-3" onClick={() => this.approve()}>
             Approve
           </Button>
-          <Button className="p-2 m-2" onClick={() => this.reject()}>
+          <Button className="p-2 m-3" onClick={() => this.reject()}>
             Reject
           </Button>
+          <div className="flex-grow-1" />
         </div>
-      );
-    return (
-      <div>
-        <Text type="p">Select a proposal</Text>
-      </div>
+      </BoxGrid>
     );
   }
 
+  renderSideCopy() {
```

Comment:
I see what you mean about being frustrated with having to create new custom views. The other problem is if you break these out into other components then you start to have a really big directory tree that could get unwieldy. I feel like I've seen example structures where you can have subdirectories by views. Might be worth looking into at some point but not a priority right now.